File: CyberLife_Calendar/src/db/pojo/EventSchedule.java
package db.pojo;

import java.util.Arrays;

public class EventSchedule {

	private int cod_repeticao;
	private int intervalo;
	private boolean[] dias_semana;
	private int fk_evento;
	
	public int getCod_repeticao() {
		return cod_repeticao;
	}
	public void setCod_repeticao(int cod_repeticao) {
		this.cod_repeticao = cod_repeticao;
	}
	public int getIntervalo() {
		return intervalo;
	}
	public void setIntervalo(int intervalo) {
		this.intervalo = intervalo;
	}
	public boolean[] getDias_semana() {
		return dias_semana;
	}
	public void setDias_semana(boolean[] dias_semana) {
		this.dias_semana = dias_semana;
	}
	public String getDias_semanaToString() {
		
		String list = "";
		
		for(int i = 0; i < this.dias_semana.length; i++) {
			
			list += (dias_semana[i] ? 1 : 0) + (i + 1 < dias_semana.length ? ",":"");
		}
		
		return list;
	}
	public void setDias_semanaToIntArray(String dias_semana) {
		
		for (int i = 0; i < this.dias_semana.length; i++) {
			this.dias_semana[i] = Boolean.parseBoolean(dias_semana.split(",")[i]);
		}
	}
	public int getFk_evento() {
		return fk_evento;
	}
	public void setFk_evento(int fk_evento) {
		this.fk_evento = fk_evento;
	}
}

File: API/JSON_objects/event/EventSchedule.php
<?php
    class EventSchedule implements \JsonSerializable {

        private $cod_repeticao;
        private $intervalo;
        private $dias_semana;
        private $fk_evento;

        public function get_cod_repeticao() : int {
            return $this->cod_repeticao;
        }

        public function set_cod_repeticao(int $cod_repeticao){
            $this->cod_repeticao = $cod_repeticao;
        }

        public function get_intervalo() : int {
            return $this->intervalo;
        }

        public function set_intervalo(int $intervalo){
            $this->intervalo = $intervalo;
        }

        public function get_dias_semana() : string {
            return $this->dias_semana;
        }

        public function set_dias_semana(string $dias_semana){
            $str_array = preg_split("/,/",$dias_semana);
            $bool_array = array();

            foreach ($str_array as $value) {
                $bool_array[] = filter_var($value, FILTER_VALIDATE_BOOLEAN);
            }

            $this->dias_semana = $bool_array;
        }

        public function get_fk_evento() : int {
            return $this->fk_evento;
        }

        public function set_fk_evento(int $fk_evento){
            $this->fk_evento = $fk_evento;
        }

        public function jsonSerialize()
        {
            $vars = get_object_vars($this);
            return $vars;
        }
    }
?>